charts/loki-distributed/values.yaml

```
@@ -943,6 +943,9 @@ gateway:
       main '$remote_addr - $remote_user [$time_local]  $status '
               '"$request" $body_bytes_sent "$http_referer" '
               '"$http_user_agent" "$http_x_forwarded_for"';
+    # -- Configure the server listener(s), this needs to be customised for IPv6 only clusters
+    serverListens: |-
+      listen             8080;
```

I think we should make it clear that this stanza is for IPv6, and IP6Enabled to enable / disable it (default is false).

```suggestion
    IP6serverListens: |-
      listen             8080;

    # -- Enable IPv6 only clusters configure
    IP6Enabled: false
```

---

@zanhsieh the current use of this block is for IP family so I could get behind using `IpFamilyServerListens` but I really don't see the benefit of that form over `serverListens`. Could you explain in a bit more detail why you think we need to explicitly tie the block to the current implementation while the listens could be modified for reasons other than just switching IP family?

---

@zanhsieh why don't we add more comments and leave it as it is, for testing purposes you can just prove that if the user leaves it alone it's a no-op? AFAIK the rest of the Nginx config is like this.

I'd have loved to be able to provide a higher level IP family argument to hide the complexity from the end users but this isn't possible as the Loki Distributed chart [won't start on IPv6 cluster](https://github.com/grafana/helm-charts/issues/157) without a hack (see https://github.com/grafana/dskit/pull/185 for the actual fix) and even when it's running it doesn't [actually work](https://github.com/grafana/loki/issues/5578).